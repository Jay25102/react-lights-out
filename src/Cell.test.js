import { fireEvent, render } from "@testing-library/react";
import Cell from "./Cell";

describe("rendering cells", function() {
    let container;

    beforeEach(function() {
        container = document.body.appendChild(document.createElement("tr"));
    });

    it("renders a cell without crashing", function() {
        render(<Cell/>, {container});
    });

    it("matches snapshot when lit", function() {
        const { asFragment } = render(<Cell isLit />, { container });
        expect(asFragment()).toMatchSnapshot();
    });

    it("matches snapshot when not lit", function() {
        const { asFragment } = render(<Cell />, { container });
        expect(asFragment()).toMatchSnapshot();
    });
});