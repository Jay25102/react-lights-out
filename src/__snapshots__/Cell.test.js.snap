// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering cells matches snapshot when lit 1`] = `<DocumentFragment />`;

exports[`rendering cells matches snapshot when not lit 1`] = `<DocumentFragment />`;
